packages/myst-cli/src/transforms/images.ts

```diff
@@ -196,257 +195,254 @@ export async function transformImages(
   );
 }
 
-type ConversionFn = (session: ISession, svg: string, writeFolder: string) => Promise<string | null>;
+const DEFAULT_IMAGE_EXTENSIONS = [ImageExtensions.png, ImageExtensions.jpg, ImageExtensions.jpeg];
+
+type ConversionOpts = {
+  file: string;
+  inkscapeAvailable: boolean;
+  imagemagickAvailable: boolean;
+};
+
+type ConversionFn = (
+  session: ISession,
+  source: string,
+  writeFolder: string,
+  conversionOpts: ConversionOpts,
+) => Promise<string | null>;
+
+/**
+ * Factory function for all simple imagemagick conversions
+ */
+function imagemagickConvert(to: ImageExtensions, from: ImageExtensions) {
+  return async (session: ISession, source: string, writeFolder: string, opts: ConversionOpts) => {
+    const { imagemagickAvailable } = opts;
+    if (imagemagickAvailable) {
+      return imagemagick.convert(to, from, session, source, writeFolder);
+    }
+    return null;
+  };
+}
+
+/**
+ * svg -> pdf using inkscape
+ */
+async function svgToPdf(
+  session: ISession,
+  source: string,
+  writeFolder: string,
+  opts: ConversionOpts,
+) {
+  const { file, inkscapeAvailable, imagemagickAvailable } = opts;
+  if (inkscapeAvailable) {
+    return inkscape.convert(ImageExtensions.svg, ImageExtensions.pdf, session, source, writeFolder);
+  } else if (imagemagickAvailable) {
+    addWarningForFile(
+      session,
+      file,
+      'To convert .svg images to .pdf, you must install inkscape.',
+      'warn',
+      { note: 'Images converted to .png as a fallback using imagemagick.' },
+    );
+  }
+  return null;
+}
+
+/**
+ * svg -> png using inkscape or imagemagick
+ */
+async function svgToPng(
+  session: ISession,
+  source: string,
+  writeFolder: string,
+  opts: ConversionOpts,
+) {
+  const { inkscapeAvailable } = opts;
+  if (inkscapeAvailable) {
+    return inkscape.convert(ImageExtensions.svg, ImageExtensions.png, session, source, writeFolder);
+  }
+  return imagemagickConvert(ImageExtensions.svg, ImageExtensions.png)(
+    session,
+    source,
+    writeFolder,
+    opts,
+  );
+}
+
+/**
+ * gif -> png using imagemagick
+ */
+async function gifToPng(
+  session: ISession,
+  source: string,
+  writeFolder: string,
+  opts: ConversionOpts,
+) {
+  const { imagemagickAvailable } = opts;
+  if (imagemagickAvailable) {
+    return imagemagick.extractFirstFrameOfGif(session, source, writeFolder);
+  }
+  return null;
+}
+
+/**
+ * These are all the available image conversion functions
+ *
+ * Get the function to convert from one extension to another with
+ * conversionFnLookup[from][to]
+ */
+const conversionFnLookup: Record<string, Record<string, ConversionFn>> = {
```

Comment:
When implementing a new conversion, you just need to implement the conversion function itself then add it here.